### PR TITLE
Enable Upstream Testing

### PR DESCRIPTION
**Pull request by bryanculver**
- Updated Dockerfile to be compatible with upstream testing
- Added workflow fragment to trigger testing
- Nautobot minimal version updated to 1.4
- `poetry.lock` regenerated

### CODE REVIEW

File: .github/workflows/upstream_testing.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+name: "Nautobot Upstream Testing"
+
+on: # yamllint disable-line rule:truthy rule:comments
+  schedule:
+    - cron: "0 4 */2 * *"  # every other day at midnight
+  push:  # yamllint disable rule:empty-values
+# TODO: REMOVE PUSH LINE ABOVE
```
Comment (bryanculver) [inline .github/workflows/upstream_testing.yml:8]:
```suggestion
```
Remove before merging.

**Comment by bryanculver** [comment]
`next` is expected to fail because it goes to a new major version of Nautobot so can be safely ignored for now.

**Comment by snaselj** [comment]
Rebased on top of `develop`

**Comment by snaselj** [comment]
Nautobot minimal version updated to 1.4

**Comment by snaselj** [comment]
This PR is ready for review now.

**Review by davidban77 — APPROVED** [review]
LGTM

**Comment by bryanculver** [comment]
> LGTM - thank you for this PR!

🌮 🥳

**Comment by bryanculver** [comment]
@davidban77 can I get a merge? :D

**Pull request merged**